### ISSUE TITLE
filetype fuse_mount2 rclone

### ISSUE THREAD

**Issue by helmecke** [vifm/vifm#610]
Hi, I was trying to mount Google Drive using rclone fuse2 and a filetype in vifm.

```vim
filetype *.drive
       \ {Mount with rclone}
       \ FUSE_MOUNT2|rclone mount --daemon %PARAM %DESTINATION_DIR,
```
I created a file `test.drive`:

```
drive_name:/
```
Opening does mount the Google Drive but the view is empty, if i switch to the other pane and execute `:cd %D` both views refresh and all files and folders are shown. I tried adding `%CLEAR` but it only refreshes the view prior not after the mount ist complete.
Do you have any idea how to fix this?

```
$ vifm --version
Version: 0.11
Git info: built out of repository
Compiled at: Oct  7 2020 19:22:04

Support of extended keys is on
Parsing of .desktop files is enabled
Without GTK+ library
With magic library
Without X11 library
Without dynamic loading of X11 library
With file program
With -n option for cp and mv
With remote command execution
```
```
$ rclone --version
rclone v1.53.2
- os/arch: linux/amd64
- go version: go1.15.3
```

**Comment by PRESFIL** [comment]
Same behavior with `rclone`. But I hit <kbd>Ctrl-l</kbd> to redraw view. May by `vifm` lose *mount inotify event*...

**Comment by xaizek** [comment]
I guess mounting happens after command has ended. You could try adding `&& usleep {some small amount}`.

@PRESFIL, I don't think there is a mount event. But it's true that detecting mounting is an issue: same path before and after mounting refers to different file-systems. inotify descriptor created for whatever was in the directory before mount won't know anything about mounting. Maybe checking inode information would help.

**Comment by helmecke** [comment]
Thank you @xaizek. Adding `sleep` does help and it's a sufficient solution for me. Feel free to close this.
```vim
filetype *.drive
       \ {Mount with rclone}
       \ FUSE_MOUNT2|rclone mount --daemon %PARAM %DESTINATION_DIR && sleep 1,
```

**Comment by PRESFIL** [comment]
Works for me too!

**Comment by PRESFIL** [comment]
Interesting stuff. I set `inotifywait` to *mountmoint* and run my filetype handler to see what's happens in moment of mounting:

```
$  inotifywait -m /tmp/kek/
Setting up watches.
Watches established.
/tmp/kek/ OPEN,ISDIR
/tmp/kek/ ACCESS,ISDIR
/tmp/kek/ CLOSE_NOWRITE,CLOSE,ISDIR
/tmp/kek/ OPEN,ISDIR
/tmp/kek/ ACCESS,ISDIR
/tmp/kek/ ACCESS,ISDIR
/tmp/kek/ CLOSE_NOWRITE,CLOSE,ISDIR
/tmp/kek/ OPEN,ISDIR
/tmp/kek/ ACCESS,ISDIR
/tmp/kek/ CLOSE_NOWRITE,CLOSE,ISDIR
```

As i understand there, is no way to create **right fix** because there is no *change state events*?

**Comment by xaizek** [comment]
> As i understand there, is no way to create right fix because there is no change state events?

Yes. If `inotify` doesn't supply this information, we need to determine it by some other means.

**Comment by PRESFIL** [comment]
`inotify` cannot return mount events, but there are smart unportable [hack][1] 😆
And seems in already [used][2]. I do not know if this is needed for `vifm`. Just leave it there.

[1]: https://superuser.com/questions/124987/hook-into-linux-mount-umount-event/847519#847519
[2]: http://stefan.buettcher.org/cs/fschange/

**Comment by PRESFIL** [comment]
Thanks!
